### ISSUE TITLE
Breaking API changes in the latest release

### ISSUE THREAD

**Issue by steelbrain** [atom-community/linter#906]
In the latest release we have some API breaking changes, You can find them in the changelog. The most significant are

* Deprecated self-APIs `Linter::{onDidChangeMessages, onDidChangeProjectMessages, getProjectMessages, setProjectMessages, deleteProjectMessages}` have been removed
* `.icon-right::before` css selector has been replaced by `.linter-gutter` of the new linter gutter element.

The last change means you'll have to update your css selectors for custom message types, they won't work in the gutter anymore, 'cause to avoid conflicts with other packages and some weird ui problems on the latest version of atom, we've decided to take a route that we should've taken a long time ago, we've decided to add our own gutter to the editor, it's position and visibility are configurable. You will have to update your package APIs to make sure they don't break with this upgrade.
TL;DR: Custom Message type css selectors have changed for the gutter.

I am waiting on your go before publishing an update.

/cc @k2b6s9j @arcanemagus @SpainTrain @basarat @radicaled @sindresorhus @iam4x @lierdakil @devoncarew @david-driscoll

I know I have missed a lot of people, but please ping them here if you know they have a linter dependent package that could be using custom error message types.

**Comment by steelbrain** [comment]
Please note that the release hasn't been pushed yet, I'll wait for you guys to update your packages before pushing a release.

**Comment by sindresorhus** [comment]
:+1: [atom-linter-xo](https://github.com/sindresorhus/atom-linter-xo) doesn't use any of those methods, so should be good.

**Comment by david-driscoll** [comment]
https://github.com/OmniSharp/omnisharp-atom should be good, I just double checked, doesn't look like we're using any of those api's.

**Comment by SpainTrain** [comment]
:+1: 

Opened issues on `AtomLinter` projects that will need to update.

**Comment by devoncarew** [comment]
:+1: Thanks for the heads up! We're good w/ the latest in master.

**Comment by Arcanemagus** [comment]
As far as I know none of the linters I've worked on use any of these functionalities.

**Comment by lierdakil** [comment]
Thanks for the heads-up, give me till Wednesday, not much time right now.
28 сент. 2015 г. 19:52 пользователь "Landon Abney" <notifications@github.com>
написал:

**Comment by basarat** [comment]
:+1: atom-typescript doesn't use those APIs :rose:

**Comment by mashaal** [comment]
Thanks, I updated https://github.com/mashaal/wild-cherry to use the new linter-gutter class 🌞🌞

**Comment by deprint** [comment]
I updated [https://github.com/deprint/build-tools-cpp](https://github.com/deprint/build-tools-cpp). Thanks for the heads up.

**Comment by lierdakil** [comment]
Relevant Atom-Haskell packages updated.

**Comment by k2b6s9j** [comment]
I will take care of this, this weekend.

**Comment by steelbrain** [comment]
I've upgraded the linter org packages depending on those css selectors, only a few are left and they'll update soon. Pushing a release now.

**Comment by steelbrain** [comment]
Congratulations on v1.7.0 people :tada:

**Comment by steelbrain** [comment]
Be advised that in the release, you can also specify a `name` attribute to your linter object, it'll help users distinguish between messages from different linters, see #908 for more info.
